tools/building.py

```diff
@@ -1063,34 +1049,15 @@ def handle_final_wasm_symbols(wasm_file, symbols_file, debug_info):
 
 
 def is_ar(filename):
+  """Return True if a the given filename is an ar archive, False otherwise.
+  """
   try:
-    if _is_ar_cache.get(filename):
-      return _is_ar_cache[filename]
     header = open(filename, 'rb').read(8)
-    sigcheck = header in (b'!<arch>\n', b'!<thin>\n')
-    _is_ar_cache[filename] = sigcheck
```

Comment:
Why does this remove the caching?

Reply:
Sorry, separate change I guess.

This function is only used in single place, which is to scan the inputs files left to right.   So it would only effect folks that specify the same library many times on the link command line.. which would be very odd.   It doesn't really have any cost either, its just reading 8 bytes from the file.

I think maybe this is a remnant from the past when we were maybe running `nm` on all these archive..

Reply:
Ah, that makes sense, yeah.